retrieve-image-tags/retrieve-image-tags.py

```diff
@@ -73,26 +85,97 @@ def _get_image_tags(registry, namespace, package, nexturl, page=1):
     result.raise_for_status()
     data = result.json()
     # Return the correct data based on the registry we queried (or paginate if registry/query requires it)
-    if registry == "dockerhub":
-        tags = [tag["name"] for tag in data["results"]]
+    if registry == 'dockerhub':
+        tags = [tag['name'] for tag in data['results']]
         #print(f"tags for {namespace}/{package}: {tags}")
-        if not data["next"]:
+        if not data['next']:
             return tags
         return tags + _get_image_tags(registry, namespace, package, "", page=page + 1)
-    elif registry == "quay.io":
-        tags = [tag["name"] for tag in data["tags"]]
-        if not data["has_additional"]:
+    if registry == 'quay.io':
+        tags = [tag['name'] for tag in data['tags']]
+        if not data['has_additional']:
             return tags
         return tags + _get_image_tags(registry, namespace, package, "", page=page + 1)
-    elif registry == "registry.k8s.io" or registry == "registry.suse.com" or registry == "gcr.io":
-        return data["tags"]
-    elif registry == "ghcr.io":
-        tags = data["tags"]
-        if not "next" in result.links:
+    if registry in ['registry.k8s.io', 'registry.suse.com', 'gcr.io']:
+        return data['tags']
+    if registry == 'ghcr.io':
+        tags = data['tags']
+        if not 'next' in result.links:
             return tags
-        return tags + _get_image_tags(registry, namespace, package, result.links["next"]["url"], page=page)
+        return tags + _get_image_tags(registry, namespace, package, result.links['next']['url'], page=page)
+    print(f"Unrecognized registry: {registry}")
+    sys.exit(1)
+
+# Function to search textfile content
+def _image_tag_already_exist(filetext, image, tag):
+    matches = re.findall(fr"{image}\s.*?\s{tag}\n", filetext)
+    if matches:
+        return True
+    return False
+
+# Function to retrieve images from helm chart using helm template
+def _extract_unique_images_from_helm_template(repo_name, chart, helm_args=''):
```

Comment:
We have several charts that use the kubernetes version to determine which images to use; making use of this requires setting `--kube-version` on the `helm template` command line. Could we add a field for that, similar to the `helmArgs` field that is used to set chart values?

Reply:
Done